routemaster/state_machine/utils.py

```diff
@@ -151,18 +174,15 @@ def labels_to_retry_for_action(
 
     active_participants = select((
         ranked_transitions.c.label_name,
-        labels.c.metadata,
     )).where(and_(
-        ranked_transitions.c.new_state == action.name,
+        ranked_transitions.c.new_state == gate.name,
         ranked_transitions.c.rank == 1,
         ranked_transitions.c.label_name == labels.c.name,
         labels.c.state_machine == state_machine.name,
+        labels.c.metadata_triggers_processed == false(),
```

Comment:
Could this be spelled something like `not_(labels.c.metadata_triggers_processed)`? (I'm not familiar enough with the sqlalchemy API to know for sure, but IMO that would read more nicely)

Reply:
71609a749670678123199080508a557010b0472f